CONTRIBUTING.md

```diff
@@ -1,44 +1,197 @@
+### Reporting bugs
+
+*Bug* is an error, design flaw, failure or fault in Iroha that causes it to produce an incorrect or unexpected result, or to behave in unintended ways. 
+
+Bugs are tracked as [GitHub Issues](https://guides.github.com/features/issues/). To submit a bug, create new Issue and include these details:
+- **Title**
+    - Write prefix `[Bug]` for the title
+    - Use a clear and descriptive title
+- **Body** - include the following sections:
+    - Steps to reproduce
+    - Expected behavior
+    - Actual behavior
+
+
+
+### Suggesting Enhancements
+
+An *enhancement* is a code or idea, which makes **existing code or design** faster, more stable, portable, secure or better in any other way.
+
+Enhancements are tracked as [GitHub Issues](https://guides.github.com/features/issues/). To submit new enhancement, create new Issue and incllude these details:
+
+- **Title**
+    - Write prefix `[Enhancement]`
+    - Use a clear and descriptive title
+- **Body** - include the following sections:
+    - *Motivation* - why do we need it?
+    - *Target* - what is going to be improved?
+    - *Description* - how to implement it?
+
+
+
+### Asking Questions
+
+A *question* is any discussion that is typically neigher a bug, nor feature requests, nor improvements - "How do I do X?".
+
+Questions are tracked as [Github Issues](https://guides.github.com/features/issues/) or via private messages in  [your favourite messenger](#contace-developers).
+
+To submit new question in GitHub Issues, it must include these details:
+
+- **Title**
+    - Write prefix `[Question]`
+    - Use a clear and descriptive title
+- **Body** - describe your question with as many details as possible.
+
+
+
+### Your First Code Contribution
+
+Read our [C++ Style Guide](#c++-style-guide) and start with beginner-friendly issues with label [`[good-first-issue]`](https://github.com/hyperledger/iroha/issues?q=is:open+is:issue+label:good-first-issue ). 
+
+
+
+### Pull Requests
+
+- Fill in [the required template](.github/PULL_REQUEST_TEMPLATE.md)
+
+- **Write tests** for new code.
+
+- Every pull request should be reviewed and **get at least two approves**.
+
+- Do not include issue numbers in the PR title or commit messages.
+
+- Include issue numbers in Pull Request body only.
+
+- When finished work, **rebase onto base branch** with 
+    ```bash
+    $ git fetch
+    $ git rebase -i <base-branch>
+    ```
+
+- Follow the [C++ Style Guide](#C++-style-guide).
+
+- Follow the [Git Style Guide](#git-commit-messages) .
+
+- **Document new code** based on the [Documentation Styleguide](#documentation-styleguide)
+
+- End all files with a newline.
+
+
+
+
+## Styleguides
+
+### Git Style Guide
+
+- **Use present tense** ("Add feautre", not "Added feature").
+- **Use imperative mood** ("Deploy docker to..." not "Deploys docker to...").
+- Write meaningful commit message.
+- **Sign every commit** with [DCO](https://github.com/apps/dco): `Signed-off-by: $NAME <$EMAIL>`. 
```

Comment:
Maybe add guide what to do if contributor messed up with signing?

Reply:
Good point.

Reply:
There is actually link to DCO, it has tutorial and description.